src/Raven.Server/Documents/Handlers/DocumentReplicationHandler.cs

```diff
@@ -1,16 +1,190 @@
+                        ["SourceDatabaseName"] = item.SourceDatabaseName,
+                        ["SourceMachineName"] = item.SourceMachineName,
+                        ["SourceUrl"] = item.SourceUrl
+                    });
+                }
+
+                var outgoing = new DynamicJsonArray();
+                foreach (var item in Database.DocumentReplicationLoader.OutgoingConnections)
+                {
+                    outgoing.Add(new DynamicJsonValue
+                    {
+                        ["Url"] = item.Url,
+                        ["Database"] = item.Database,
+                        ["Disabled"] = item.Disabled,
+                        ["IgnoredClient"] = item.IgnoredClient,
+                        ["SkipIndexReplication"] = item.SkipIndexReplication,
+                        ["SpecifiedCollections"] = item.SpecifiedCollections
+                    });
+                }
+
+                context.Write(writer, new DynamicJsonValue
+                {
+                    ["IncomingConnections"] = incoming,
+                    ["OutgoingConnections"] = outgoing
+                });
+            }
+            return Task.CompletedTask;
+        }
+
+        [RavenAction("/databases/*/replication/debug/outgoing-failures", "GET")]
+        public Task GetReplicationOugoingFailureStats()
+        {
+            HttpContext.Response.StatusCode = (int)HttpStatusCode.OK;
+            DocumentsOperationContext context;
+            using (ContextPool.AllocateOperationContext(out context))
+            using (var writer = new BlittableJsonTextWriter(context, ResponseBodyStream()))
+            {
+                var data = new DynamicJsonArray();
+                foreach (var item in Database.DocumentReplicationLoader.OutgoingFailureInfo)
+                {
+                    data.Add(new DynamicJsonValue
+                    {
+                        ["Key"] = new DynamicJsonValue
+                        {
+                            ["Url"] = item.Key.Url,
+                            ["Database"] = item.Key.Database,
+                            ["Disabled"] = item.Key.Disabled,
+                            ["IgnoredClient"] = item.Key.IgnoredClient,
+                            ["SkipIndexReplication"] = item.Key.SkipIndexReplication,
+                            ["SpecifiedCollections"] = item.Key.SpecifiedCollections
+                        },
+                        ["Value"] = new DynamicJsonValue
+                        {
+                            ["ErrorCount"] = item.Value.ErrorCount,
+                            ["NextTimout"] = item.Value.NextTimout
+                        }
+                    });
+                }
+
+                context.Write(writer, data);
+            }
+            return Task.CompletedTask;
+        }
+
+        [RavenAction("/databases/*/replication/debug/incoming-last-activity-time", "GET")]
+        public Task GetReplicationIncomingActivityTimes()
```

Comment:
All of those endpoints can be summed into a single `/replication/debug/status` one.
It would be easier to work with